### PR TITLE
Fix Travis build pip install issue in pre installation step

### PR DESCRIPTION
**Pull request by neerajmangal**
## Related issue and scope

- [ ] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [ ] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/openwhisk/blob/master/CONTRIBUTING.md).
- [ ] I reviewed the [style guides](https://github.com/apache/openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

**Comment by BillZong** [comment]
It's been a replicated fix with #4679 .

**Comment by neerajmangal** [comment]
Pip issue is solved by adding "--user" flag and adding python3-pip. Travis builds are failing now with this issue. 

```bash
> Task :swagger-java-client:compileJava
/home/travis/build/apache/openwhisk/tests/build/swagger-code-java/src/main/java/io/swagger/client/ApiException.java:19: error: package javax.annotation does not exist
@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaClientCodegen", date = "2019-10-11T06:14:22.400Z")
                 ^
```

Seems like the default version of java is picked as 11. 
```bash
java -Xmx32m -version
openjdk version "11.0.2" 2019-01-15
OpenJDK Runtime Environment 18.9 (build 11.0.2+9)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.2+9, mixed mode)
$ javac -J-Xmx32m -version
javac 11.0.2
```
and it's causing the issue with the swagger doc compilation issue above due to the removal of javax libs. 

Fix - 

1. Added dist as xenial 
2. Pin JDK version to openjdk8
3. Added python3-pip package

For future stability, I think we should completely move to python version 3.x as currently, our build, test system is integrated with python2.7 which will be deprecated in Jan 2020. - Raised another issue for the same. 

Please let me know your feedback.

File: .travis.yml [hunk 1]
```diff
@@ -58,8 +59,9 @@ cache:
 
 before_install:
   - ./tools/travis/docker.sh
-  - pip install --upgrade pip setuptools six
-  - pip3 install --upgrade pip setuptools six
+  - sudo apt-get install -y python3-pip
```
Comment (chetanmeh) [inline .travis.yml]: @neerajmangal Does the build still fails with `--upgrade` flag as current pip version seems to be old

```
You are using pip version 8.1.1, however version 19.2.3 is available.
```

**Comment by neerajmangal** [comment]
All test passed, one system test is failing. 

```bash
services.KafkaConnectorTests > Kafka connector should send and receive a kafka message even after shutdown one of instances STANDARD_OUT
    expected exit code = 0
    exit code = 1
    stdout: 
    stderr: Cannot connect to the Docker daemon at tcp://172.17.0.1:4243. Is the docker daemon running?
services.KafkaConnectorTests > Kafka connector should send and receive a kafka message even after shutdown one of instances FAILED
```

Looking at it (thanks @chetanmeh ), found that docker version is not honoring `DOCKER_OPTS`. 
We are working on a fix to use `daemon.json` file https://docs.docker.com/config/daemon/systemd/#custom-docker-daemon-options which will have all the `DOCKER_OPTS` in it.

File: .travis.yml (hunk 1 above)
Comment (rabbah) [reply .travis.yml]: @dgrove-oss do these help on your pr?

Comment (dgrove-oss) [reply .travis.yml]: I didn't have time to finish yesterday; this one looks closer than mine.  I will close mine in favor of this one.

**Comment by rabbah** [comment]
Might be transient - I restarted the failed job.

**Comment by rabbah** [comment]
nm i didnt read the comment carefully.

File: .travis.yml [hunk 2]
```diff
@@ -18,7 +18,8 @@
 sudo: required
 
 group: deprecated-2017Q3
```
Comment (rabbah) [inline .travis.yml]: can we drop this deprecated group?

Comment (dgrove-oss) [reply .travis.yml]: I think yes, since it is selecting a `trusty` variant and we are attempting to switch to `xenial`

File: .travis.yml (hunk 1 above)
Comment (neerajmangal) [reply .travis.yml]: > @neerajmangal Does the build still fails with `--upgrade` flag as current pip version seems to be old
> 
> ```
> You are using pip version 8.1.1, however version 19.2.3 is available.
> ```

It is fixed now using --upgrade.

**Comment by rabbah** [comment]
Thanks @neerajmangal for tackling this. If it's a single test failing, we could ignore it temporarily to unblock the other PRs and then address the docker issue. Wasn't clear to me if it was more than one test though.

**Comment by neerajmangal** [comment]
> Thanks @neerajmangal for tackling this. If it's a single test failing, we could ignore it temporarily to unblock the other PRs and then address the docker issue. Wasn't clear to me if it was more than one test though.

One single test related to `Kafka connector should send and receive a Kafka message even after shutdown one of instances` was failing, but it leads to the root cause of docker issue on xenial which may in future cause failures. The docker issue is fixed now. All is well now :).

**Comment by chetanmeh** [comment]
Thanks @neerajmangal for digging through the rabbit hole here and getting the build back to green 🎉

**Comment by chetanmeh** [comment]
Looks like few other module builds are also failing with similar error related to pip upgrade. Like https://travis-ci.org/apache/openwhisk-cli/builds

Probably other module whose status is shown as error in https://github.com/apache/openwhisk/blob/master/docs/dev/modules.md

**Comment by neerajmangal** [comment]
> Looks like few other module builds are also failing with similar error related to pip upgrade. Like https://travis-ci.org/apache/openwhisk-cli/builds
> 
> Probably other module whose status is shown as error in https://github.com/apache/openwhisk/blob/master/docs/dev/modules.md

Hi @chetanmeh, I will pick up a few other modules tomorrow to fix.

**Pull request merged**